src/bin/hq.rs

```diff
@@ -283,9 +308,8 @@ async fn command_worker_stop(
     opts: WorkerStopOpts,
 ) -> anyhow::Result<()> {
     let mut connection = get_client_connection(&gsettings.server_directory()).await?;
-    stop_worker(&mut connection, opts.worker_id)
-        .await
-        .map_err(|e| e.into())
+    stop_worker(&mut connection, WorkerSelector::from(opts.selector)).await?;
```

Comment:
It's more common to use `opts.selector.into()` than `WorkerSelector::from(opts.selector)`, but that's a detail.